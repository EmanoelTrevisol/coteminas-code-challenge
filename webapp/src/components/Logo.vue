<template>
  <h1 class="logo">
    mmartan
  </h1>
</template>

<script>
export default {};
</script>

<style lang="stylus" scoped>
h1 {
	margin: 0 10px 0;
	color: $brand-logo;
	font-weight: 500;
	letter-spacing: 1.25px;
	width: 150px;
}
</style>
