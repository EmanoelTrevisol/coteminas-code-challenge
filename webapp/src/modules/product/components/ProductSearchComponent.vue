<template>
  <div class="product-search-input">
    <font-awesome-icon icon="search" />
    <input
      class="search-input"
      type="text"
      placeholder="Buscar produtos"
      :value="filter"
      @input="(evt) => setSearchFilter(evt.target.value)"
    />
    <font-awesome-icon icon="times-circle" />
  </div>
</template>

<script>
import { debounce } from 'lodash';
import { mapState } from 'vuex';

export default {
  components: {},
  data() {
    return {};
  },
  computed: {
    ...mapState('product', {
      filter: (state) => state.filter,
    }),
  },
  methods: {
    setSearchFilter(val) {
      this.$store.dispatch('product/updateSearchFilter', val);
      this.getList();
    },
    getList: debounce(function() {
      this.$store.dispatch('product/getList');
    }, 600),
  },
};
</script>

<style lang="stylus" scoped>
.product-search-input {
	display: flex;
	align-items: center;
	justify-content: space-between;
	border: 1px solid $black;
	padding: 10px 20px;
	width: 450px;
	border-radius: 20px;

	.search-icon {
		height: 20px;
	}

	.search-input {
		height: 20px;
		border: none;
	}
}
</style>
