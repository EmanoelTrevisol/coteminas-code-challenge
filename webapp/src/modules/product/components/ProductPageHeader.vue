<template>
  <div class="product-page-header">
    <logo />
    <product-search-component />
  </div>
</template>

<script>
import Logo from '@/components/Logo';
import ProductSearchComponent from './ProductSearchComponent';

export default {
  components: {
    Logo,
    ProductSearchComponent,
  },
};
</script>

<style lang="stylus" scoped>
.product-page-header {
	display: flex;
	justify-content: space-between;
	align-items: center;
	width: 100%;
}
</style>
