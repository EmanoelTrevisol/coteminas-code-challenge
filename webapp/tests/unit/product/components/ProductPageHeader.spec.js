import { shallowMount } from '@vue/test-utils';
import ProductPageHeader from '@/modules/product/components/ProductPageHeader';
import ProductSearchComponent from '@/modules/product/components/ProductSearchComponent';
import Logo from '@/components/Logo';

describe('ProductPageHeader.vue', () => {
  it('renders component', () => {
    const wrapper = shallowMount(ProductPageHeader);

    expect(wrapper.text()).toMatchSnapshot();
  });

  it('renders logo and search', () => {
    const wrapper = shallowMount(ProductPageHeader);

    const logo = wrapper.findComponent(Logo);
    const productSearch = wrapper.findComponent(ProductSearchComponent);

    expect(logo.exists()).toBe(true);
    expect(productSearch.exists()).toBe(true);
  });
});
