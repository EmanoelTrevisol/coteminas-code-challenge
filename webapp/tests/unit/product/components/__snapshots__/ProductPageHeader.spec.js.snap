// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPageHeader.vue renders component 1`] = `""`;
